### ISSUE TITLE
Accessible non-virtual destructor

### ISSUE THREAD

**Issue by clementperon** [oneapi-src/oneTBB#1318]
Hi,

I tried to compiled the latest oneTBB 2021.11.0 and got the following error:

```
./obj/dev/_deps/onetbb-src/src/tbb/../../include/tbb/../oneapi/tbb/detail/_task.h:214:31: error: base class ‘class tbb::detail::d1::task_traits’ has accessible non-virtual destructor [-Werror=non-virtual-dtor]
  214 | class alignas(task_alignment) task : public task_traits {
      |                               ^~~~
```

I'm using GCC 11.4.0

**Comment by dnmokhov** [comment]
Thank you for reporting this, @clementperon.

Could you please let us know your platform, CMake configuration, and/or compilation command?

**Comment by clementperon** [comment]
@dnmokhov, 

Ok I'm trying to make a minimal example and see If I can reproduce it easily.

**Comment by clementperon** [comment]
@dnmokhov 

After investigation this issue is due too my project that is built with -Werror and -Weffc++

I have made a simple example of failure here:
https://github.com/clementperon/OneTBB-Example
https://github.com/clementperon/OneTBB-Example/actions/runs/8137780296/job/22236966999

**Comment by dnmokhov** [comment]
Thank you for providing the example, @clementperon.

Having a pointer to `tbb::detail::d1::task_traits` is not intended, so the class has no virtual destructor.

Since this is an external dependency for your project, you could consider using the `-isystem` directory search option for oneTBB.

E.g., using the standard `-I` option:

> $ g++ -c -std=c++17 -Werror -Weffc++ main.cpp **_-I_** oneTBB-2021.11.0/include
> In file included from oneTBB-2021.11.0/include/oneapi/tbb/parallel_for.h:23,
>                  from main.cpp:3:
> oneTBB-2021.11.0/include/oneapi/tbb/detail/_task.h:214:31: error: base class ‘class tbb::detail::d1::task_traits’ has accessible non-virtual destructor [-Werror=non-virtual-dtor]
>   214 | class alignas(task_alignment) task : public task_traits {

Switching to `-isystem` suppresses the warnings:

> $ g++ -c -std=c++17 -Werror -Weffc++ main.cpp **_-isystem_** oneTBB-2021.11.0/include

**Comment by dnmokhov** [comment]
Please reopen if the issue remains.
